src/common/memContext.h

```diff
@@ -198,25 +198,26 @@ nearest try block.
 Use the MEM_CONTEXT*() macros when possible rather than reimplement the boilerplate for every memory context block.
 ***********************************************************************************************************************************/
 // Create a new mem context in the current mem context. The new context must be either kept with memContextKeep() or discarded with
-// memContextDisard() before the parent context can be popped off the stack.
+// memContextDisard() before switching back from the parent context.
 MemContext *memContextNew(const char *name);
 
-// Push a switchable type mem context onto the stack which will then become the current mem context. Only context of type switch can
-// become the current mem context.
+// Switch to a context making it the current mem context
 void memContextSwitch(MemContext *this);
 
-// Pop a mem context off the stack and make the prior (first switchable mem context) the current mem context. If the top of the
-// stack is a type new mem context (new context must be kept or discarded) then an error will be thrown in debug builds.
+// Switch back to the context that was current before the last switch.  This is the same context that will be returned from
+// memContextPrior().  If the last function called was memContextNew(), then memContextKeep()/memContextDiscard() must be called
```

Comment:
If we say it's the same as Prior than it makes me wonder why we even have SwitchBack. I'd rather not confuse the issue and would prefer we just remove that sentence.